### DISCUSSION TITLE
What does `shell --fancy` do?

### DISCUSSION THREAD

**Discussion by isaacl** [pypa/pipenv#5472] [Q&A]
What exactly does `pipenv shell --fancy` do?

Also, I notice the omz [pipenv plugin](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/pipenv/pipenv.plugin.zsh) does not use it.  Should it have an option?

**Comment by matteius** [comment]
From reading the code, for windows, it is always "fancy" and other systems the flag determines whether we invoke the shell with the logic of `shell.fork(*fork_args)` and otherwise it tries `shell.fork_compat(*fork_args)` and if that fails falls back to `shell.fork(*fork_args)`.  I'd like if eventually we can deprecate the `--fancy` argument and consolidate the shell fork logic into a single code path, it seems close already.

**Reply by isaacl** [reply]
My read was the `shell.fork` execs into a shell process, while `shell.fork_compat` stays involved in every keystroke using pexpect.

So in theory, `shell.fork` should be more efficient, especially for large amounts of output.

Related [pyenv patch](https://github.com/pyenv/pyenv/pull/2526) -- to support using `--fancy` with `pyenv`.


However I still haven't gotten `--fancy` to fully work.
